### PR TITLE
Add futures for ref and inout arguments with default values (issue #7924)

### PR DESCRIPTION
**Pull request by cassella**
Add futures for issue #7924

### CODE REVIEW

**Comment by cassella** [comment]
Is this the right way to write a ``.bad`` file for an internal error?

I didn't add futures for the versions where the call is made with a var for an actual, in which case the compiler doesn't complain about the argument having a default value.  I think it should complain in that case too, as the presence of the default value suggests the function can be called with that argument omitted.

I'm not attached to the phrasing of the error I used in the ``.good`` files.

**Comment by lydia-duncan** [comment]
> I didn't add futures for the versions where the call is made with a var for an actual, in which case the compiler doesn't complain about the argument having a default value. I think it should complain in that case too, as the presence of the default value suggests the function can be called with that argument omitted.

This falls under the general category of us only generating errors for structures that are used at compilation, which is something we are aware of but haven't worked on

File: test/functions/default-arguments/default-argument-inout.chpl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+proc foo(inout x = 7) { x = 17; writeln(x); }
+foo();
+
```
Comment (lydia-duncan) [inline test/functions/default-arguments/default-argument-inout.chpl]: I don't like that these tests have an extra line at the end of them, but I'm willing to live with it.  That said, I'm going to wait until Monday to merge so if you get motivated... :p

File: test/functions/default-arguments/default-argument-ref.bad [hunk 2]
```diff
@@ -0,0 +1,8 @@
+default-argument-ref.chpl:2: internal error: EXP1418 chpl version 1.18.0 pre-release (a1d1fb9)
+Note: This source location is a guess.
+
+Internal errors indicate a bug in the Chapel compiler ("It's us, not you"),
+and we're sorry for the hassle.  We would appreciate your reporting this bug -- 
+please see https://chapel-lang.org/bugs.html for instructions.  In the meantime,
+the filename + line number above may be useful in working around the issue.
+
```
Comment (lydia-duncan) [inline test/functions/default-arguments/default-argument-ref.bad]: Yeah, this is right.

File: test/functions/default-arguments/default-argument-inout.chpl (hunk 1 above)
Comment (cassella) [reply test/functions/default-arguments/default-argument-inout.chpl]: Motivation found.  Here y'go.

**Pull request merged**